src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -801,9 +801,7 @@ public static int Compare(object first, object second, bool ignoreCase, IFormatP
                 }
             }
 
-            string firstString = first as string;
-
-            if (firstString != null)
+            if (first is string firstString)
             {
                 string secondString = second as string;
                 if (secondString == null)
```

Comment:
@iSazonov I think we should be careful about following one pattern too strictly. I think `as string` is better here because the `if` condition is `secondString == null`.  Changing it to `if (!(second is string secondString))` would make it less readable, which goes against our goal.

Reply:
Using different patterns (especially next to each other) also reduces readability - I just stumbled over it.
My suggestion was just to roll back the change at all to old pattern or use one pattern.